src/track/external-thumbnails-handler.js

```diff
@@ -0,0 +1,263 @@
+
+  /**
+   * the player instance.
+   * @type {Player}
+   * @private
+   */
+  _player: Player;
+
+  /**
+   * event manager for the external thumbnails handler
+   * @type {EventManager}
+   * @private
+   */
+  _eventManager: EventManager;
+
+  /**
+   * the processed thumbnail cues
+   * @type {Array<Cue>}
+   * @private
+   */
+  _cues: Array<PKThumbnailVttCue> = [];
+
+  /**
+   * base url of the images or the sprite img
+   * @type {string}
+   * @private
+   */
+  _imgBaseUrl: string;
+
+  /**
+   * computed img dimensions based on its natural ratio
+   * @type {Object}
+   * @private
+   */
+  _computedImgSize: {width:number, height: number} = {width: null, height: null};
+
+  /**
+   * start the loading and parsing process of the vtt thumbnails file.
+   * @returns {void}
+   * @public
+   */
+  async load(): Promise<void> {
+    const thumbnailsConfig: PKExternalThumbnailsConfig = this._player.sources.thumbnails;
+    if (!thumbnailsConfig) {
+      return [];
+    }
+    this._imgBaseUrl = this._player.sources.thumbnails.imgBaseUrl || null;
+    await this._downloadAndParseCues(thumbnailsConfig);
+  }
+
+  /**
+   * find and return the img info for the requested time.
+   * @param {number} time - time point in th playback timeline in milliseconds.
+   * @returns {ThumbnailInfo | null} - the thumbnail img info for the requested time.
+   * @public
+   */
+  getThumbnail(time: number): ThumbnailInfo | null {
+    const cue: PKThumbnailVttCue = this._findCue(time);
+    if (cue) {
+      let {size, coordinates, imgUrl} = cue;
+      size = size ? size : this._computedImgSize;
+      const thumbnailInfo = {url: imgUrl, ...size, ...coordinates};
+      return new ThumbnailInfo(thumbnailInfo);
+    }
+    return null;
+  }
+
+  /**
+   * indicate whether or not this player using external vtt thumbnails.
+   * @returns {boolean} whether or not this player using external vtt thumbnails.
+   * @public
+   */
+  isUsingVttThumbnails(): boolean {
+    return !!this._cues.length;
+  }
+
+  /**
+   * download and parse the vtt file
+   * @param {PKExternalThumbnailsConfig} thumbnailsConfig - the thumbnails user config
+   * @returns {Promise<void>} - resolve when the loading and parsing process is complete
+   * @private
+   */
+  async _downloadAndParseCues(thumbnailsConfig: PKExternalThumbnailsConfig): Promise<void> {
+    const VttStr: string = await this._downloadVttFile(thumbnailsConfig);
+    const cuesArray: Array<Cue> = await this._processVtt(VttStr);
+    this._cues = await this._formatIntoThumbnailCues(cuesArray);
+  }
+
+  /**
+   * Make a request to download the vtt file.
+   * @param {PKExternalThumbnailsConfig} thumbnailsConfig - the thumbnails config object.
+   * @returns {Promise<string>} - resolves with the vtt string.
+   * @private
+   */
+  async _downloadVttFile(thumbnailsConfig: PKExternalThumbnailsConfig): Promise<string> {
+    return await Utils.Http.execute(thumbnailsConfig.vttUrl, {}, 'GET');
```

Comment:
add debug printout in case download fails

Reply:
added error handling which print the error